### PR TITLE
Suggest matching environment independent of capitalization

### PR DESCRIPTION
**Pull request by kir0ul**
This PR fixes https://github.com/openai/gym/issues/2455.

### CODE REVIEW

**Comment by jkterry1** [comment]
Please use classic control, toy text or Atari environments for the tests (others will have some changes in the near future)

**Comment by jkterry1** [comment]
(also tests are failing)

**Review by JesseFarebro — CHANGES_REQUESTED** [review]
We can get something more robust by using the standard library `difflib`. In particular, you can do,

```py
import difflib

matches = difflib.get_close_matches(id, self.env_specs.keys())
```

This will apply to more than just lower-case correction as is implemented in this PR. If the list of matches is non-zero you could just format the list and provide this to the user.

**Comment by kir0ul** [comment]
> We can get something more robust by using the standard library `difflib`. 

Thanks for the suggestion, I didn't know about `difflib`.

File: tests/envs/test_registration.py [hunk 1]
```diff
@@ -24,11 +26,30 @@ def __init__(self, arg1, arg2, arg3):
 
 
 def test_make():
-    env = envs.make("CartPole-v0")
-    assert env.spec.id == "CartPole-v0"
+    env = envs.make("CartPole-v1")
```
Comment (JesseFarebro) [inline tests/envs/test_registration.py:29]: Could you revert these changes to `-v0`?

File: gym/envs/registration.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
-import re
-import sys
+import contextlib
```
Comment (JesseFarebro) [inline gym/envs/registration.py:1]: Could you revert the `isort` changes here?

File: tests/envs/test_registration.py [hunk 3]
```diff
@@ -89,7 +110,7 @@ def test_missing_lookup():
 def test_malformed_lookup():
     registry = registration.EnvRegistry()
     try:
-        registry.spec(u"“Breakout-v0”")
+        registry.spec("“Breakout-v0”")
```
Comment (JesseFarebro) [inline tests/envs/test_registration.py:113]: Could you revert this change? We can include this in a different PR where we clean up these issues.

File: tests/envs/test_registration.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # -*- coding: utf-8 -*-
+import pytest
```
Comment (JesseFarebro) [inline tests/envs/test_registration.py:2]: Could you revert this `isort` change?

**Review by JesseFarebro — CHANGES_REQUESTED** [review]

File: tests/envs/test_registration.py (hunk 3 above)
Comment (kir0ul) [reply tests/envs/test_registration.py:113]: The thing is, if I put back the `u` before the string, the `black` pipeline fails, so I'd leave it this way.

**Comment by kir0ul** [comment]
Oops, was working on the wrong branch and pushed your changes here instead of in https://github.com/openai/gym/pull/2535. Will redo it in the right branch.

**Comment by kir0ul** [comment]
I think we can close this since all the code and tests have been copied in https://github.com/openai/gym/pull/2535.
